tests/fixtures/workers/helpers/start-and-seed-server.ts

```diff
@@ -77,14 +80,38 @@ const startAndSeedServer = async () => {
 
   ;(axios.defaults.headers as any).Authorization = `Bearer ${api_key}`
 
-  const [schlageLock, augustLock, connectWebview] = await Promise.all([
-    addFakeSchlageDevices(axios),
-    addFakeAugustDevices(axios),
+  const providersMap = {
+    schlage: () => addFakeSchlageDevices(axios),
+    august: () => addFakeAugustDevices(axios),
+    minut: () => addFakeMinutDevices(axios),
+  }
+  const [connectWebview, ...devicesByProvider] = await Promise.all([
     axios.post("/connect_webviews/create", {
-      accepted_providers: ["schlage", "august"],
+      // TODO: remove filter when minut is ready
+      accepted_providers: load_devices_from.filter(
+        (provider) => provider !== "minut"
+      ),
     }),
+    ...load_devices_from.map((provider) => providersMap[provider]()),
   ])
 
+  const devices = {
+    ...load_devices_from.reduce((acc, provider, index) => {
+      if (provider === "schlage") {
+        acc["schlageLock"] = devicesByProvider[index]
+        return acc
+      }
+
+      if (provider === "august") {
+        acc["augustLock"] = devicesByProvider[index]
+        return acc
+      }
+
+      acc[provider] = devicesByProvider[index]
+      return acc
+    }, {} as Record<"augustLock" | "minut" | "schlageLock", any>),
+  }
```

Comment:
The readability here is hard, here's why:

* There's an indexing scheme that's not clear, particularly because some of the variable names are unclear
* The type is buried in the implementation, if something starts getting complicated, better to lift the type up to the top
* `.reduce` isn't a great function because it moves the type definition towards the bottom, in practice loops tend to be preferable
* Mark deprecated or legacy behavior explicitly
* I THINK there's a spread here for no reason? Why spread the reduce

Principally, we should try to avoid letting optimizations like `Promise.all` impact readability.

```suggestion
  const devices: Record<keyof providersMap & 'schlageLock' | 'augustLock', any> = {}
  
  for (let i = 0; i < load_devices_from.length; i++) {
     const provider = load_devices_from[i]
     devices[provider] = devicesByProvider[i]
  }
  
  // TODO: remove when we deprecate schlageLock and augustLock
  if (devices.august) {
    devices.augustLock = devices.august
  }
  if (devices.schlage) {
    devices.schlageLock = devices.schlage
  }
```